enzyme/Enzyme/DifferentialUseAnalysis.cpp

```diff
@@ -64,15 +64,33 @@ bool DifferentialUseAnalysis::is_use_directly_needed_in_reverse(
     return false;
   }
 
-  if (isa<LoadInst>(user)) {
-    if (EnzymeRuntimeActivityCheck &&
-        TR.query(const_cast<llvm::Instruction *>(user))[{-1}].isFloat() &&
-        !gutils->isConstantInstruction(const_cast<llvm::Instruction *>(user))) {
-      if (EnzymePrintDiffUse)
-        llvm::errs() << " Need direct primal of " << *val
-                     << " in reverse from runtime active load " << *user
-                     << "\n";
-      return true;
+  if (auto LI = dyn_cast<LoadInst>(user)) {
+    if (EnzymeRuntimeActivityCheck) {
+      auto vd = TR.query(const_cast<llvm::Instruction *>(user));
+      if (!vd.isKnown()) {
+        auto ET = LI->getType();
+        // From reverse mode type
+        if (looseTypeAnalysis || true) {
```

Comment:
It verbatim needs to replicate the same behavior as adjointgenerator:
https://github.com/EnzymeAD/Enzyme/blob/194875cbccd73d63cacfefbfa85c1f583c2fa1fe/enzyme/Enzyme/AdjointGenerator.h#L556

That true historically was there because without it various tests failed (and that was the previous behavior). Type Analysis being upgraded sufficiently can enable that true to disappear.

Reply:
Can you please add that comment at least here?